src/openrct2.c

```diff
@@ -294,6 +294,43 @@ void openrct2_launch()
 			break;
 		case STARTUP_ACTION_OPEN:
 			assert(gOpenRCT2StartupActionPath != NULL);
+
+			// A path that includes a colon is illegal on Windows, so it is almost certainly a URL
```

Comment:
In that case you can check for `://`. That isn't valid in EXT and prevents edge cases like you mentioned.